packages/core/src/browser/xhrProxy.spec.ts

```diff
@@ -209,4 +209,26 @@ describe('xhr proxy', () => {
       },
     })
   })
+
+  it('should track only the second request (the first request is canceled)', (done) => {
```

Comment:
Do we have test cases for the other issues/changes:
```
We can have a max stack size error when using a single instance of XMLHTTPRequest.
Some requests can be missed due to the override of onreadystatechange by the user
Attach our event listeners sooner in the XMLHTTPRequest lifecycle
Ensure to attach our listeners only once per request.
```

Reply:
Not really, I'll make more unit tests to cover these cases.

Reply:
The new version of the PR only focus on the first issue: We can have a max stack size error when using a single instance of XMLHTTPRequest.